### ISSUE TITLE
Forwarding logs to stdout broken in `1.12.x`

### ISSUE THREAD

**Issue by diist** [aws/amazon-vpc-cni-k8s#2222]
**What happened**:
The new Go entrypoint `cmd/aws-vpc-cni/main.go` does not allow forwarding `./aws-k8s-agent` (`ipamd`) logs to `stdout` when configured to do so.

This line does not work in Go (as no shell is interpreting the command) https://github.com/aws/amazon-vpc-cni-k8s/blob/master/cmd/aws-vpc-cni/main.go#L373:
```go
ipamdDaemon := exec.Command(cmd, "|", "tee", "-i", agentLogPath, "2>&1")
```

Here's the current workaround I'm using:
```diff
    agentLogPath := getEnv(envAgentLogPath, defaultAgentLogPath)

    cmd := "./aws-k8s-agent"
-   ipamdDaemon := exec.Command(cmd, "|", "tee", "-i", agentLogPath, "2>&1")
+   ipamdDaemon := exec.Command(cmd)
+   ipamdDaemon.Stdout = os.Stdout
+   ipamdDaemon.Stderr = os.Stderr
    err = ipamdDaemon.Start()
    if err != nil {
        log.WithError(err).Errorf("Failed to execute command: %s", cmd)
```

This is not a permanent fix as lower in that file the `agentLogPath` (which is not written anymore) is read if `ipamd` fails to start.

**Attach logs**
The logs... there are no logs from `ipamd`!

**What you expected to happen**:
Seeing the debug logs from `ipamd`

**How to reproduce it (as minimally and precisely as possible)**:
Set the env to:
```
AWS_VPC_K8S_CNI_LOGLEVEL=Debug
AWS_VPC_K8S_CNI_LOG_FILE=stdout
```
And run `./aws-vpc-cni`

**Anything else we need to know?**:
Slightly unrelated: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/cmd/aws-vpc-cni/main.go#L352 can never print because you're using straight logrus but not setting the log level (defaults to info).

**Environment**:
- Kubernetes version (use `kubectl version`): `1.22`
- CNI Version: `1.12.1`
- OS (e.g: `cat /etc/os-release`): `public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-iptables:latest.2`
- Kernel (e.g. `uname -a`): unknown

**Comment by jdn5126** [comment]
Ah thank you for catching this @diist ! We'll have to figure out the right golang logic here to fix this. If you'd like to continue working on this, that would be awesome! Otherwise, I'll mark this as a good first issue and someone will pick it up when they have a chance.

**Comment by anushkabishnoi** [comment]
Hello, this is my first experience with open source contributions. Would really appreciate it if you explained exactly what is needed to be done.

I would love to work on this issue.

Thanks in advance! :)

**Comment by jdn5126** [comment]
@diist I think there is a bit of confusion here, as `agentLogPath` comes from the environment variable `AGENT_LOG_PATH`, not `AWS_VPC_K8S_CNI_LOG_FILE`. `AGENT_LOG_PATH` is used for stdout and stderr from the IPAMD process (`aws-k8s-agent`), but the agent itself logs primarily to the file specified by `AWS_VPC_K8S_CNI_LOG_FILE`.

Still, the pipe to `tee` is not currently working, so we should fix this.

**Comment by jdn5126** [comment]
@anushkabishnoi thank you for the interest! The problem here is that the latest VPC CNI image no longer contains a shell, so the `tee` executable is not present and pipes cannot be interpreted.

The goal for this function call is to redirect stdout and stderr to a custom file. So the change would be to set `Cmd.Stdout` and `cmd.Stderr` to the value of `agentLogPath` (reference https://pkg.go.dev/os/exec#Cmd). If `agentLogPath` is the string "stdout", then you need to redirect both to `os.Stdout`.

You would need to test this on a running cluster by introducing ways for IPAM process to fail. Since there is no shell in the container, you can run `docker cp <container_id>:/app/<file> .` on the node and inspect the log.

**Comment by jdn5126** [comment]
Closed as fixed by #2237
